src-v2/Judge/NTS.Judge/Adapters/Behinds/StartlistBehind.cs

```diff
@@ -9,37 +9,25 @@
 
 namespace NTS.Judge.Adapters.Behinds;
 
-public class StartlistBehind : ObservableBehind, IStartupInitializer, IStartlistBehind
+public class StartlistBehind : ObservableBehind, IStartlistUpcoming, IStartlistHistory
 {
     readonly IRepository<Participation> _participationRepository;
 
     public StartlistBehind(IRepository<Participation> participationRepository)
     {
         _participationRepository = participationRepository;
-        var startlist = new StartList();
+        var startlist = new StartList(EmitChange);
         Startlist = startlist;
     }
 
-    public StartList Startlist { get; set; }
-    public List<Start> Upcoming => Startlist.Upcoming;
-    public List<Start> History => Startlist.History;
+    StartList Startlist { get; set; }
```

Comment:
Favor private fields. Use properties when
- protected, ie. has to be used internally by a child type
- public - we don't never ever want public fields
- private - in limited scenarios when they cannot be avoided, such as when using the [Inject] attribute.

Reply:
Applied the changes but adhered to private fields as you used the verb **favor**. Could you please elaborate on the clarification that private should be used only when it cannot be avoided. As it is phrased now it seems to me that we have the following options:
1. protected - use only when property is used in child type
2. private - avoid if you can
3. never declare public fields as it is probably a security vulnerability, I am not sure about the reasoning

Reply:
I don't quite understand your question, but it seems to me that you're not making the distinction between a field and a property. Or at least you seem to be confusing them. While I think that on a lower programming level all class members are essentially fields the major distinction between a field and a property is that _fields hold state, while properties expose an API to the consumers (provide tool that can granularly configure access - getters and setters_. 

A Property has the additional advantage as being recognized by IDEs which provide useful tools for tracking their usage:
![image](https://github.com/user-attachments/assets/9c12d3b3-2eb5-4c03-9991-b46992c95bc8)

It is not considered safe to expose fields, because they almost always have to be modifier internally in the class, but there is no way to restrict their modification from outside of the class. With properties you can expose only a getter. The only case when fields should be exposed is when they are `const` or `static readonly` as these two combinations of modifiers ensure that the value of this field cannot be changed.